test/model/recent_senders_test.dart

```diff
@@ -0,0 +1,201 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/recent_senders.dart';
+import '../example_data.dart' as eg;
+import 'recent_senders_checks.dart';
+
+void checkMatchesMessages(List<Message> messages, RecentSenders model) {
+  final Map<int, Map<int, MessageIdTracker>> expectedStreamSenders = {};
+  final Map<int, Map<String, Map<int, MessageIdTracker>>> expectedTopicSenders = {};
+  for (final message in messages) {
+    if (message is! StreamMessage) continue;
```

Comment:
Clearer if this throws on a DM, rather than ignore it. The test cases shouldn't be passing DMs among the expected messages — if they do, the test case will be confusing because it may look to the reader as if the DM is expected to be there in the data structure.